pkg/homedir/homedir_unix.go

```diff
@@ -25,17 +37,105 @@ func Key() string {
 //
 // If needing to do nss lookups, do not disable cgo or set osusergo.
 func Get() string {
-	home := os.Getenv(Key())
-	if home == "" {
-		if u, err := user.Current(); err == nil {
-			return u.HomeDir
+	homeDirOnce.Do(func() {
+		homeDir := os.Getenv(Key())
```

Comment:
This looks dangerous as `homeDir` will always evaluate to the inner scope and the package-level variable will never be set.